src/data.rs

```diff
@@ -48,11 +52,22 @@ impl Incoming<'_> {
         }
     }
 
+    /// Is this a CLOSE frame?
+    // TODO: return `Option<CloseReason>`? Is this even needed?
+    pub fn is_close(&self) -> bool {
+        if let Incoming::Closed(_) = self {
+            true
+        } else {
+            false
+        }
+    }
+
     /// The length of data (number of bytes).
     pub fn len(&self) -> usize {
         match self {
             Incoming::Data(d) => d.len(),
-            Incoming::Pong(d) => d.len()
+            Incoming::Pong(d) => d.len(),
+            Incoming::Closed(reason) => std::mem::size_of_val(reason), // TODO: this makes no sense; use `unimplemented!()` here?
```

Comment:
This method is indeed a bit questionable.
I don't see a situation where you would want the length of the payload of `Pong` and treat it the same way as the length of the payload of `Data`.

I'd go with removing the method entirely.